internal/server/source_manager.go

```diff
@@ -295,6 +298,17 @@ func (s *sourceManager) stop(ctx context.Context) {
 	close(s.closed)
 }
 
+func (s *sourceManager) removeMetrics() {
+	if s.rep.Metrics() != nil {
+		registry := s.rep.Metrics()
+		registry.RemoveGauge(s.lastSnapshotStartTime)
```

Comment:
That seems wrong-each of these calls would remove all gauges instead of just gauges for a particular source.

Reply:
Should be ok. Tested in e2e, only the gauge with the correct labels will be deleted.